bindings/rust/src/lib.rs

```diff
@@ -159,17 +216,20 @@ mod tests {
         let result = unsafe { instance.unsafe_execute(1, &[], 0) };
         assert!(!result.trapped);
         assert!(result.has_value);
-        unsafe {
-            assert_eq!(result.value.i64, 42);
-        } // FIXME: this is actually i32
+        assert_eq!(result.value.as_i32(), 42);
 
+        // Explicit type specification
         let result =
-            unsafe { instance.unsafe_execute(2, &[Value { i64: 42 }, Value { i64: 2 }], 0) };
+            unsafe { instance.unsafe_execute(2, &[(42 as i32).into(), (2 as i32).into()], 0) };
+        assert!(!result.trapped);
+        assert!(result.has_value);
+        assert_eq!(result.value.as_i32(), 21);
+
+        // Implicit i64 types (even though the code expects i32)
+        let result = unsafe { instance.unsafe_execute(2, &[42.into(), 2.into()], 0) };
```

Comment:
What is this `.into()` doing?

Reply:
Explicitly invoking the `from` trait.

Reply:
Converts integers to `Value`, where `Value` is deduced from function parameter type, I guess